lib/spack/spack/fetch_strategy.py

```diff
@@ -829,9 +829,15 @@ def fetch(self):
             args.append('at revision %s' % self.revision)
         tty.msg("Trying to clone Mercurial repository:", self.url, *args)
 
-        args = ['clone', self.url]
+        args = ['clone']
+
+        if spack.insecure:
+            args.append('--insecure')
```

Comment:
I took a look at `hg_fetch.py` but it's gibberish to me. Which line actually does the fetching?